insights/parsers/sctp.py

```diff
@@ -254,3 +257,72 @@ def search(self, **args):
                 when args search do not match then it returns `[]`.
         """
         return keyword_search(self.data, **args)
+
+
+@parser(Specs.sctp_snmp)
+class SCTPSnmp(Parser, dict):
+    """
+    This parser parses the content of ``/proc/net/sctp/snmp`` file,
+    which contains statistics related to SCTP states, packets and chunks.
+
+    Sample content::
+
+        SctpCurrEstab                   	5380
+        SctpActiveEstabs                	12749
+        SctpPassiveEstabs               	55
+        SctpAborteds                    	2142
+        SctpShutdowns                   	5295
+        SctpOutOfBlues                  	36786
+        SctpChecksumErrors              	0
+        SctpOutCtrlChunks               	1051492
+
+    Data is stored in a dictionary.
+
+    Examples:
+
+        >>> type(sctp_snmp)
+        <class 'insights.parsers.sctp.SCTPSnmp'>
+        >>> sctp_snmp.get('SctpCurrEstab')
+        5380
+        >>> sctp_snmp.get('SctpChecksumErrors') == 0
+        True
+        >>> sctp_snmp.stats
+        {'SctpCurrEstab': 5380, 'SctpActiveEstabs': 12749, 'SctpPassiveEstabs': 55, 'SctpAborteds': 2142, 'SctpShutdowns': 5295, 'SctpOutOfBlues': 36786, 'SctpChecksumErrors': 0, 'SctpOutCtrlChunks': 1051492}
+
+    Resultant Data::
+
+        {
+            'SctpCurrEstab': 5380,
+            'SctpActiveEstabs': 12749,
+            'SctpPassiveEstabs': 55,
+            'SctpAborteds': 2142,
+            'SctpShutdowns': 5295,
+            'SctpOutOfBlues': 36786,
+            'SctpChecksumErrors': 0,
+            ...
+            ...
+        }
+
+    Raises:
+        SkipException: When contents are empty.
+        ParseException: When file contents are not in expected format.
+    """
+
+    def parse_content(self, content):
+        if (not content) or (not self.file_path):
+            raise SkipException("No Contents")
+
+        for line in content:
+            line_strip = line.split()
+            if len(line_strip) != 2:
+                raise ParseException("Contents are not compatible to this parser")
+
+            self[line_strip[0]] = int(line_strip[1]) if line_strip[1].isdigit() else None
```

Comment:
As you did this check `line_strip[1].isdigit()`, is it necessary to raise another `ParseException` when the value is not a digit?

Reply:
@xiangce  Raised exception here in case there are more than 2 elements after splitting. Just making sure file is in correct format. 
Should I remove this?

Reply:
@Rohan21Lobo , it depends on you. From my view, it seems incorrect to set the value to `None` when the second splits is not a digit. But if you have confidence that it (non-digit) will never happen, better remove this check.

Reply:
@Rohan21Lobo - You have already checked that there are 2 values after unpack, so no need to confirm it again. Also the second value will not be empty, it will always be digit. So, please remove the check.

Reply:
Done 
https://github.com/RedHatInsights/insights-core/blob/c4b9e132e82631c774998115df6cdc1403c962d4/insights/parsers/sctp.py#L319-L324